### ISSUE TITLE
Props

### ISSUE THREAD

**Issue by rohrlach** [upstage-org/limelight#11]
**Acceptance Criteria**
- [ ] A 'Prop' resource will be created
- [ ] The Prop resource should virtually mimic the Avatar resource from an architectural perspective 
- [ ] Logged in users can create, edit, delete Props
- [ ] Logged in users can assign/unassign props from stages.

**Comment by rohrlach** [comment]
@helenvarley can you create acceptance criteria for how the props should work during a performance?

**Comment by helenvarley** [comment]
yes; i think that the prop should NOT virtually mimic the avatar, otherwise it's really not that useful to have a separate resource. it would be good to be able to place props directly on the stage as independent objects, not something that has to be held by an avatar (as it is currently). 
something like:
- player clicks on prop icon to "hold", then clicks on stage to "place"; prop is then visible on the stage, as a static object.
- player clicks on prop icon a second time to remove it from the stage.
- or, player clicks on prop icon a second time then clicks somewhere else on stage to make the prop move to that point.
- if that works then perhaps it could be click twice on the icon to remove it from the stage.
- need a way also to animate.

it would be good to have some discussion with others eg @vickis & mem, about how props could be redesigned to work better.

**Comment by rohrlach** [comment]
Yes I agree.  What I mean by 'virtually mimic' is that, in terms of the server side representation, there is actually very little that separates a representation of media types - How they behave client side is a different matter entirely :)

With that being said, I have a feeling I could do better architecturally in this regard, so I'm putting this into my backlog for further (ie: broader-scoped) consideration.

**Comment by vickis** [comment]
yes like the prop being more 'object like’ i do like to ability to ’share' & carry props but then the prop could perhaps be placed OR held 
for example some things you might wish to pass around

BUT for example o walk through a forest, jump from cloud to cloud or swim through strands of seaweed would be interesting to be able to do 
(i guess like what we did with the trees prop in babayaga (and also that petyr et al did in their work with the ‘apparition’ in the corridor …

like it!

**Comment by helenvarley** [comment]
i agree vicki. shall we write a list of how we'd like props to work?
- can be placed on stage & left there (independently of an avatar)
- can be moved on the stage (independently of an avatar)
- can be held & carried by an avatar (i.e. it moves along when you move the avatar)
- can be passed from one avatar to another (which we can currently do)
- avatars can go in front of or behind props
- can be animated
- can be rotated?
... more?

**Comment by helenvarley** [comment]
in UpStage, props are visual objects that can only be put on stage when they are held by an avatar, and they stay at the top left of the avatar and move around the stage with the avatar. this is not always that useful - could be done with another frame in the avatar for example.(or draw)

    - it would be useful to be able to place a prop on the stage, independently of any avatar, & have it stay there until removed (it would also be useful for avatars to be able to move between propos (like layers)

    - this could also be done with an avatar (although clear might remove them - if props were enabled, placed and cleared separately that would be ideal)

    - what is the difference between avatars & props? props don't have voices or names that appear on stage, they aren't held or move around; (if a propo s attached to an avatar it can move) if able to be placed without the avatar it remains imovable 

    thinking of how props worked in the palace: you could have multiple instances of the same prop object on a stage, eg you have one beer in the prop bar, but you can place it onto the stage multiple times


    props work ike backdrops except that they exist on layers so avatars have the ability to move between them (be behind) and come out to infront of 

    and they are/can be smaller & different shapes than backdrops
